modules/tinymce/src/core/main/ts/html/InvalidNodes.ts

```diff
@@ -159,17 +160,31 @@ const hasClosest = (node: AstNode, parentName: string): boolean => {
   return false;
 };
 
-const isInvalid = (schema: Schema, node: AstNode, parent: AstNode | null | undefined = node.parent): boolean => {
+const isInvalid = (schema: Schema, node: AstNode, isInsert: boolean, parent: AstNode | null | undefined = node.parent): boolean => {
+  if (!parent) {
+    return false;
+  }
+
   // Check if the node is a valid child of the parent node. If the child is
   // unknown we don't collect it since it's probably a custom element
-  if (parent && schema.children[node.name] && !schema.isValidChild(parent.name, node.name)) {
+  if (schema.children[node.name] && !schema.isValidChild(parent.name, node.name)) {
     return true;
+  }
+
   // Anchors are a special case and cannot be nested
-  } else if (parent && node.name === 'a' && hasClosest(parent, 'a')) {
+  if (node.name === 'a' && hasClosest(parent, 'a')) {
     return true;
-  } else {
-    return false;
   }
+
+  // heading element is valid if it is the only one child of summary (though on inserting it is still invalid)
+  if (AstNodeType.isHeading(node) && hasClosest(node, 'summary')) {
```

Comment:
Maybe be specific here and only check for headings as a direct child of the summary not any parent that is summary that the `hasClosest` will match. However not sure if that is ever going to happen.

Reply:
When we insert `h2` in the `summary` that already contains `h1`, then the closest parent would be `h1` but we still need to handle this case to correctly merge headings.